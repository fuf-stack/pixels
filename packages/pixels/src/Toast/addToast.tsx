import type { TVClassName, TVProps } from '@fuf-stack/pixel-utils';
import type { ToastProps as HeroToastProps } from '@heroui/toast';
import type { ReactNode } from 'react';

import { toast as heroToastVariants } from '@heroui/theme';
import { addToast as heroAddToast } from '@heroui/toast';

import { tv, variantsToClassNames } from '@fuf-stack/pixel-utils';

export const toastVariants = tv({
  slots: {
    base: '',
    title: '',
    description: '',
    icon: '',
    loadingIcon: '',
    content: '',
    motionDiv: '',
    progressTrack: '',
    progressIndicator: '',
    closeButton: '',
    closeIcon: '',
  },
  variants: {
    // see:  https://github.com/heroui-inc/heroui/blob/canary/packages/core/theme/src/components/toast.ts
    color: {
      info: {},
      ...heroToastVariants.variants.color,
    },
    variant: {
      ...heroToastVariants.variants.variant,
    },
  },
  compoundVariants: [
    ...heroToastVariants.compoundVariants,
    {
      variant: 'solid',
      color: 'info',
      className: {
        base: 'bg-info text-info-foreground',
        title: 'text-info-foreground',
        description: 'text-info-foreground',
        icon: 'text-info-foreground',
        closeIcon: 'border-info-400 bg-info-100',
        closeButton: 'text-info-400 hover:text-info-600',
        progressIndicator: 'bg-info-400',
      },
    },
    {
      color: 'info',
      variant: 'flat',
      className: {
        base: 'bg-info-50 text-info-600 dark:bg-info-50/50',
        title: 'text-info-600',
        description: 'text-info-500',
        icon: 'text-info-600',
        closeIcon: 'border-info-400 bg-info-100',
        closeButton: 'text-info-400 hover:text-info-600',
        progressIndicator: 'bg-info-400',
      },
    },

    {
      color: 'info',
      variant: 'bordered',
      className: {
        base: 'border-small border-info text-info-600',
        title: 'text-info-600',
        description: 'text-info-500',
        icon: 'text-info-600',
        closeIcon: 'border-info-400 bg-info-100',
        closeButton: 'text-info-400 hover:text-info-600',
        progressIndicator: 'bg-info-400',
      },
    },
  ],
});

export type AddToastVariantProps = TVProps<typeof toastVariants>;
type ClassName = TVClassName<typeof toastVariants>;

export interface ToastProps extends AddToastVariantProps {
  /** CSS class name */
  className?: ClassName;
  /** Icon displayed at the end of the Alert */
  closeIcon?: ReactNode;
  /** Color of the alert */
  color?: AddToastVariantProps['color'];
  /** Content displayed inside the Alert */
  description?: ReactNode;
  /** Content displayed at the end of the Alert */
  endContent?: ReactNode;
  /** Icon displayed at the start of the Alert */
  icon?: ReactNode;
  /** Icon or animation displayed until promise is resolved */
  loadingIcon?: ReactNode;
  /** Callback function called when the toast is closed */
  onClose?: () => void;
  /** Promise based on which the notification will be styled */
  promise?: Promise<void>;
  /** Whether the toast should show a progress bar */
  shouldShowTimeoutProgress?: boolean;
  /** Whether the toast should show a close button */
  showCloseButton?: boolean;
  /** Whether the toast should show an icon */
  showIcon?: boolean;
  /** Time the toast is displayed */
  timeout?: number;
  /** Content displayed at the top of the toast */
  title?: ReactNode;
  /** Variant of the toast */
  variant?: AddToastVariantProps['variant'];
}

/**
 * addToast function based on [HeroUI Toast](https://www.heroui.com//docs/components/toast)
 */
export const addToast = ({
  className = undefined,
  closeIcon = undefined,
  color = 'default',
  description = undefined,
  endContent = undefined,
  icon = undefined,
  loadingIcon = undefined,
  onClose = undefined,
  promise = undefined,
  shouldShowTimeoutProgress = false,
  showCloseButton = true,
  showIcon = true,
  timeout = 5000,
  title = undefined,
  variant = 'bordered',
}: ToastProps) => {
  const variants = toastVariants({
    color,
    variant,
  });
  const isHeroUIColor = Object.keys(heroToastVariants.variants.color).includes(
    color,
  );
  const classNames = variantsToClassNames(variants, className, 'base');
  heroAddToast({
    classNames,
    closeIcon,
    color: isHeroUIColor ? (color as HeroToastProps['color']) : undefined,
    // TODO: 'data-testid': testId,
    description,
    endContent,
    hideCloseButton: !showCloseButton,
    hideIcon: !showIcon,
    icon,
    loadingIcon,
    onClose,
    promise,
    shouldShowTimeoutProgress,
    timeout,
    title,
    variant,
  });
};

export default addToast;
