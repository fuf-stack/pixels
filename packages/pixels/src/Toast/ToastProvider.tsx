import type { ToastProps as HeroToastProps } from '@heroui/toast';
import type { ToastProps } from './addToast';

import { toast as heroToastVariants } from '@heroui/theme';
import { ToastProvider as HeroToastProvider } from '@heroui/toast';

import { variantsToClassNames } from '@fuf-stack/pixel-utils';

import { toastVariants } from './addToast';

export type ToastProviderToastProps = Omit<
  ToastProps,
  'description' | 'endContent' | 'title'
>;

export interface ToastProviderProps {
  disableAnimation?: boolean;
  maxVisibleToasts?: number;
  placement?:
    | 'bottom-right'
    | 'bottom-left'
    | 'bottom-center'
    | 'top-right'
    | 'top-left'
    | 'top-center';

  toastProps?: ToastProviderToastProps;
  toastOffset?: number;
  regionProps?: { classNames?: Record<'base', string> };
}

/**
 * ToastProvider component based on [HeroUI Toast](https://www.heroui.com//docs/components/toast)
 */
const ToastProvider = ({
  disableAnimation = false,
  maxVisibleToasts = 5,
  placement = 'top-center',
  toastProps = undefined,
  toastOffset = 26,
  regionProps = undefined,
}: ToastProviderProps) => {
  const variants = toastVariants({
    color: toastProps?.color,
    variant: toastProps?.variant,
  });
  const isHeroUIColor = Object.keys(heroToastVariants.variants.color).includes(
    toastProps?.color || '',
  );
  const classNames = variantsToClassNames(
    variants,
    toastProps?.className,
    'base',
  );
  return (
    <HeroToastProvider
      disableAnimation={disableAnimation}
      maxVisibleToasts={maxVisibleToasts}
      placement={placement}
      regionProps={regionProps}
      toastOffset={toastOffset}
      toastProps={{
        ...toastProps,
        classNames,
        color: isHeroUIColor
          ? (toastProps?.color as HeroToastProps['color'])
          : undefined,
      }}
    />
  );
};

export default ToastProvider;
