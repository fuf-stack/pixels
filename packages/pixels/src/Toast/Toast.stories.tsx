import type { Meta, StoryObj } from '@storybook/react';
import type { ToastProps } from './addToast';

import { action } from '@storybook/addon-actions';

import { Button } from '../Button';
import { addToast, Toast, toastVariants } from './index';
import ToastProvider from './ToastProvider';

const meta: Meta<typeof addToast> = {
  title: 'pixels/Toast',
  component: Toast,
  args: {
    onClose: action('closed'),
  },
  decorators: [
    (Story) => (
      <>
        <ToastProvider placement="top-center" />
        <Story />
      </>
    ),
  ],
};

export default meta;
type Story = StoryObj<typeof addToast>;

export const Default: Story = {
  args: {
    title: "Something's Up",
  },
  render: (args) => (
    <Button onClick={() => addToast({ ...args })}>Show success toast</Button>
  ),
};

export const AllProps: Story = {
  args: {
    className: { progressIndicator: 'moep1' },
    title: "Something's Up",
    description: 'A message of varying importance has been detected.',
    color: 'info' as ToastProps['color'],
    variant: 'bordered' as ToastProps['variant'],
    timeout: 60000,
    closeIcon: undefined,
    endContent: 'the end',
    icon: undefined,
    loadingIcon: undefined,
    shouldShowTimeoutProgress: true,
    showCloseButton: true,
    showIcon: true,
    placement: 'top-right',
  },
  render: (args) => (
    <Button onClick={() => addToast({ ...args })}>Show toast</Button>
  ),
};

export const AllColors: Story = {
  render: (args) => (
    <>
      {[...Object.keys(toastVariants.variants.color)].map((color) => (
        <div key={color} className="mb-12">
          <h2 className="mb-4 text-lg font-bold">{color}</h2>
          <Button
            onClick={() =>
              addToast({
                ...args,

                color: color as ToastProps['color'],
              })
            }
          >{`${color}`}</Button>
        </div>
      ))}
    </>
  ),
  args: {
    title: "Something's Up",
    description: 'A message of varying importance has been detected.',
    variant: 'bordered' as ToastProps['variant'],
  },
};

export const AllVariants: Story = {
  render: (args) => (
    <>
      {Object.keys(toastVariants.variants.variant).map((variant) => (
        <div key={`${variant}`} className="mb-6">
          <div className="mb-2 text-sm text-foreground">{variant}</div>
          <Button
            onClick={() =>
              addToast({
                ...args,

                variant: variant as ToastProps['variant'],
              })
            }
          >{`${variant}`}</Button>
        </div>
      ))}
    </>
  ),
  args: {
    title: "Something's Up",
    description: 'A message of varying importance has been detected.',
    color: 'info' as ToastProps['color'],
    timeout: 0,
  },
};

export const LongContent: Story = {
  args: {
    className: { progressIndicator: 'moep1' },
    title: "The Toast That Won't Shut Up (and Other Charming Qualities)",
    description:
      "Ever needed to tell your users something important, but also wanted to channel your inner toddler and just yell it at them repeatedly? Then you've come to the right place! This story showcases the HeroUI/NextUI Toast component â€“ a powerful tool for delivering notifications, alerts, and passive-aggressive messages to your unsuspecting audience. Here, you can tweak the Toast's appearance, duration, and severity (from a gentle 'heads up' to a full-blown 'CODE RED! THE COFFEE MACHINE IS EMPTY!'). Be warned: overuse can lead to user frustration and potentially feature requests for a 'mute all toasts' button. But hey, sometimes you just need to make sure they really see that discount code, right? Explore different variations, from the subtly informative to the gloriously attention-grabbing. Just remember, with great toast power comes great responsibility... or at least a well-placed 'X' button.",
    color: 'info' as ToastProps['color'],
    variant: 'bordered' as ToastProps['variant'],
    timeout: 60000,
    closeIcon: undefined,
    // description,
    endContent: 'the end',
    icon: undefined,
    loadingIcon: undefined,
    shouldShowTimeoutProgress: true,
    showCloseButton: true,
    showIcon: true,
    placement: 'top-right',
  },
  render: (args) => (
    <Button onClick={() => addToast({ ...args })}>Show toast</Button>
  ),
};
