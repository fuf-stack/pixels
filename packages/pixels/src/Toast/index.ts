import { addToast, toastVariants } from './addToast';
import ToastProvider from './ToastProvider';

export type { ToastProps, AddToastVariantProps } from './addToast';
export type { ToastProviderProps } from './ToastProvider';

export { addToast, ToastProvider, toastVariants };

export default ToastProvider;
