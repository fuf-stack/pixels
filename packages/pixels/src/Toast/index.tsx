import { addToast, toastVariants } from './addToast';
import ToastProvider from './ToastProvider';

export type { ToastProps, AddToastVariantProps } from './addToast';

export type { ToastProviderProps } from './ToastProvider';

export { ToastProvider };

export { addToast, toastVariants };

export default ToastProvider;
