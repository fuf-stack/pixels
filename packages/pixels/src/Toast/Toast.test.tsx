import { describe } from 'vitest';

import storySnapshots from '@repo/storybook-config/story-snapshots';

import * as stories from './Toast.stories';

describe('Story Snapshots', () => {
  storySnapshots(stories);
});
