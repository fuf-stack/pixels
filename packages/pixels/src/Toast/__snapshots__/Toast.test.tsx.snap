// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Story Snapshots > AllColors 1`] = `
<div>
  <div
    class="mb-12"
  >
    <h2
      class="mb-4 text-lg font-bold"
    >
      info
    </h2>
    <button
      class="group tap-highlight-transparent data-[hover=true]:opacity-hover z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover"
      data-react-aria-pressable="true"
      tabindex="0"
      type="button"
    >
      info
    </button>
  </div>
  <div
    class="mb-12"
  >
    <h2
      class="mb-4 text-lg font-bold"
    >
      default
    </h2>
    <button
      class="group tap-highlight-transparent data-[hover=true]:opacity-hover z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover"
      data-react-aria-pressable="true"
      tabindex="0"
      type="button"
    >
      default
    </button>
  </div>
  <div
    class="mb-12"
  >
    <h2
      class="mb-4 text-lg font-bold"
    >
      foreground
    </h2>
    <button
      class="group tap-highlight-transparent data-[hover=true]:opacity-hover z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover"
      data-react-aria-pressable="true"
      tabindex="0"
      type="button"
    >
      foreground
    </button>
  </div>
  <div
    class="mb-12"
  >
    <h2
      class="mb-4 text-lg font-bold"
    >
      primary
    </h2>
    <button
      class="group tap-highlight-transparent data-[hover=true]:opacity-hover z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover"
      data-react-aria-pressable="true"
      tabindex="0"
      type="button"
    >
      primary
    </button>
  </div>
  <div
    class="mb-12"
  >
    <h2
      class="mb-4 text-lg font-bold"
    >
      secondary
    </h2>
    <button
      class="group tap-highlight-transparent data-[hover=true]:opacity-hover z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover"
      data-react-aria-pressable="true"
      tabindex="0"
      type="button"
    >
      secondary
    </button>
  </div>
  <div
    class="mb-12"
  >
    <h2
      class="mb-4 text-lg font-bold"
    >
      success
    </h2>
    <button
      class="group tap-highlight-transparent data-[hover=true]:opacity-hover z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover"
      data-react-aria-pressable="true"
      tabindex="0"
      type="button"
    >
      success
    </button>
  </div>
  <div
    class="mb-12"
  >
    <h2
      class="mb-4 text-lg font-bold"
    >
      warning
    </h2>
    <button
      class="group tap-highlight-transparent data-[hover=true]:opacity-hover z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover"
      data-react-aria-pressable="true"
      tabindex="0"
      type="button"
    >
      warning
    </button>
  </div>
  <div
    class="mb-12"
  >
    <h2
      class="mb-4 text-lg font-bold"
    >
      danger
    </h2>
    <button
      class="group tap-highlight-transparent data-[hover=true]:opacity-hover z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover"
      data-react-aria-pressable="true"
      tabindex="0"
      type="button"
    >
      danger
    </button>
  </div>
</div>
`;

exports[`Story Snapshots > AllProps 1`] = `
<div>
  <button
    class="group tap-highlight-transparent data-[hover=true]:opacity-hover z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover"
    data-react-aria-pressable="true"
    tabindex="0"
    type="button"
  >
    Show toast
  </button>
</div>
`;

exports[`Story Snapshots > AllVariants 1`] = `
<div>
  <div
    class="mb-6"
  >
    <div
      class="text-foreground mb-2 text-sm"
    >
      flat
    </div>
    <button
      class="group tap-highlight-transparent data-[hover=true]:opacity-hover z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover"
      data-react-aria-pressable="true"
      tabindex="0"
      type="button"
    >
      flat
    </button>
  </div>
  <div
    class="mb-6"
  >
    <div
      class="text-foreground mb-2 text-sm"
    >
      solid
    </div>
    <button
      class="group tap-highlight-transparent data-[hover=true]:opacity-hover z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover"
      data-react-aria-pressable="true"
      tabindex="0"
      type="button"
    >
      solid
    </button>
  </div>
  <div
    class="mb-6"
  >
    <div
      class="text-foreground mb-2 text-sm"
    >
      bordered
    </div>
    <button
      class="group tap-highlight-transparent data-[hover=true]:opacity-hover z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover"
      data-react-aria-pressable="true"
      tabindex="0"
      type="button"
    >
      bordered
    </button>
  </div>
</div>
`;

exports[`Story Snapshots > Default 1`] = `
<div>
  <button
    class="group tap-highlight-transparent data-[hover=true]:opacity-hover z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover"
    data-react-aria-pressable="true"
    tabindex="0"
    type="button"
  >
    Show success toast
  </button>
</div>
`;

exports[`Story Snapshots > LongContent 1`] = `
<div>
  <button
    class="group tap-highlight-transparent data-[hover=true]:opacity-hover z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover"
    data-react-aria-pressable="true"
    tabindex="0"
    type="button"
  >
    Show toast
  </button>
</div>
`;
